core/src/main/java/org/jdbi/v3/core/statement/ArgumentBinder.java

```diff
@@ -48,11 +49,13 @@ private static void bindPositional(int size, Binding binding, PreparedStatement
     }
 
     private static void bindNamed(List<String> parameterNames, Binding binding, PreparedStatement statement, StatementContext context) {
-        if (parameterNames.isEmpty() && !binding.isEmpty()) {
+        // TODO throw on any mismatch, not only empty/non-empty
```

Comment:
Exception to this todo is out-parameters.

Reply:
Never used those. I assume they can't be distinguished?

Reply:
Not without parsing intent out of the SQL statement, which is way outside our project charter.

Reply:
This is why missing positional parameters are just ignored in the `bindPositional` method.

Reply:
So can we do any extra validation then or is what's present as good as it can possibly be?